core/src/dht/dht_reducers.rs

```diff
@@ -131,11 +167,225 @@ pub(crate) fn reduce_add_link(
     let storage = new_store.meta_storage();
     let result = storage.write().unwrap().add_eav(&eav);
     new_store
-        .add_link_actions_mut()
-        .insert(action_wrapper.clone(), result);
+        .actions_mut()
+        .insert(action_wrapper.clone(), result.map(|_| link.base().clone()));
     Some(new_store)
 }
 
+//
+pub(crate) fn reduce_update_entry(
+    context: Arc<Context>,
+    old_store: &DhtStore,
+    action_wrapper: &ActionWrapper,
+) -> Option<DhtStore> {
+    // Setup
+    let action = action_wrapper.action();
+    let (old_address, new_address) = unwrap_to!(action => Action::UpdateEntry);
+    let mut new_store = (*old_store).clone();
+    let content_storage = &old_store.content_storage().clone();
+    // pre-condition: Must already have old_entry in local content_storage
+    if !(*content_storage.read().unwrap())
+        .contains(&old_address)
+        .unwrap()
+    {
+        new_store.actions_mut().insert(
+            action_wrapper.clone(),
+            Err(HolochainError::ErrorGeneric(String::from(
+                "old_entry is not present in DHT's CAS",
+            ))),
+        );
+        return Some(new_store);
+    }
+    //  pre-condition: Must already have new_entry in local content_storage
+    if !(*content_storage.read().unwrap())
+        .contains(&new_address)
+        .unwrap()
+    {
+        new_store.actions_mut().insert(
+            action_wrapper.clone(),
+            Err(HolochainError::ErrorGeneric(String::from(
+                "new_entry is not present in DHT's CAS",
+            ))),
+        );
+        return Some(new_store);
+    }
+    // pre-condition: old_entry's latest version must have LIVE crud-status
+    // get latest entry
+    let mut entry_result = GetEntryResult::new();
+    let res = get_entry_rec(
```

Comment:
Hm, this won't work when we switch to network code. `get_entry_rec` will need to do recursive network look-ups. That can't happen inside a reducer since it has to involve actions that are reduced as well. This (DHT) reducer should only care for the last step, i.e. if it has the entry that needs it crud modify link to be set.

`get_entry_rec` should be called in the according action creator of `update_entry`.
And that action creator should be called in network/handler.rs when processing a DhtMetaData with attribute "modify".